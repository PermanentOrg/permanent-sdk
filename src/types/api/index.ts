export * from './AccountVo';
export * from './ArchiveVo';
export * from './BaseVo';
export * from './FileVo';
export * from './FolderVo';
export * from './PresignedPost';
export * from './RecordVo';
export * from './S3UploadVo';
