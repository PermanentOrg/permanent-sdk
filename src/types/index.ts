export * from './api';
export * from './sdk';
export * from './stela';
