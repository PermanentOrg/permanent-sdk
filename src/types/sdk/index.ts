export * from "./AccessRole";
export * from "./Account";
export * from "./Archive";
export * from "./ArchiveRecord";
export * from "./ArchiveRecordType";
export * from "./ClientConfiguration";
export * from "./DerivativeType";
export * from "./File";
export * from "./FileSystemItem";
export * from "./Folder";
export * from "./ShareLink";
export * from "./Status";
