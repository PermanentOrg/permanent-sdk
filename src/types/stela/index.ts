export * from "./Items";
export * from "./PaginatedItems";
export * from "./StelaFile";
export * from "./StelaFolder";
export * from "./StelaRecord";
