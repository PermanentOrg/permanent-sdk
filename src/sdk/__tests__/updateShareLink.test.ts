import nock from 'nock';
import { updateShareLink } from '..';
import { AccessRole, Status } from '../../types';

describe('updateShareLink', () => {
  test('should update the share link', async () => {
    nock('https://permanent.local')
      .post(
        '/api/share/updateShareLink',
        {
          shareByUrlId: 1,
          maxUses: 10,
          previewToggle: false,
          autoApproveToggle: true,
          defaultAccessRole: 'access.role.editor',
          expiresDT: '2022-01-01T00:00:00.000Z',
        },
      )
      .replyWithFile(
        200,
        `${__dirname}/fixtures/updateShareLink/shareLinkVo.json`,
        { 'Content-Type': 'application/json' },
      );

    const shareLink = await updateShareLink(
      {
        bearerToken: '12345',
        baseUrl: 'https://permanent.local/api',
      },
      {
        id: 1,
        sharedFileSystemItemId: 1,
        status: Status.Ok,
        byAccountId: 1,
        url: 'https://permanent.local/share/1',
        urlToken: '1',
        maxUses: 10,
        showPreview: false,
        autoApprove: true,
        defaultAccessRole: AccessRole.Editor,
        expiresAt: new Date('2022-01-01T00:00:00Z'),
        createdAt: new Date('2021-01-01T00:00:00Z'),
        updatedAt: new Date('2021-01-01T00:00:00Z'),
      },
    );

    expect(shareLink).toMatchSnapshot();
  });
});
