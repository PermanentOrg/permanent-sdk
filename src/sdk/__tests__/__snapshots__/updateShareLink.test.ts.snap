// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updateShareLink should update the share link 1`] = `
{
  "autoApprove": false,
  "byAccountId": 1770,
  "createdAt": 2024-02-02T19:05:25.000Z,
  "defaultAccessRole": "access.role.editor",
  "expiresAt": undefined,
  "id": 203,
  "maxUses": 0,
  "sharedFileSystemItemId": 8306,
  "showPreview": false,
  "status": "status.generic.ok",
  "updatedAt": 2024-02-02T19:05:53.000Z,
  "url": "https://local.permanent.org/share/4fe0ce05ecd16d64d7fd223e66e279ea8d692d300c4ae61b0a5ed39530ff87fc",
  "urlToken": "4fe0ce05ecd16d64d7fd223e66e279ea8d692d300c4ae61b0a5ed39530ff87fc",
  "uses": 0,
}
`;
